File: backend/internal/application/ports/speaker.go
package ports

import (
	"context"

	"github.com/gdg-eskisehir/events/backend/internal/domain"
)

type SpeakerRepository interface {
	GetByID(ctx context.Context, id string) (*domain.Speaker, error)
	List(ctx context.Context, query *string) ([]*domain.Speaker, error)
	Create(ctx context.Context, s *domain.Speaker) error
	Update(ctx context.Context, id string, fullName, bio, avatarURL *string) error
}

File: apps/mobile/lib/features/speakers/domain/entities/speaker.dart
import 'package:equatable/equatable.dart';

class Speaker extends Equatable {
  const Speaker({
    required this.id,
    required this.fullName,
    this.bio,
    this.avatarUrl,
  });

  final String id;
  final String fullName;
  final String? bio;
  final String? avatarUrl;

  @override
  List<Object?> get props => [id, fullName, bio, avatarUrl];
}
